src/software/ai/navigator/path_planner/path_planner_test.cpp

```diff
@@ -0,0 +1,255 @@
+#include "software/ai/navigator/path_planner/path_planner.h"
+
+#include <gtest/gtest.h>
+
+#include <typeinfo>
+
+#include "software/ai/navigator/obstacle/obstacle.h"
+#include "software/ai/navigator/path_planner/straight_line_path_planner.h"
+#include "software/ai/navigator/path_planner/theta_star_path_planner.h"
+#include "software/new_geom/point.h"
+#include "software/new_geom/rectangle.h"
+#include "software/test_util/test_util.h"
+#include "software/world/world.h"
+
+// length at which to evaluate the spline value and check
+// if it intersects an obstacle
+static constexpr double PATH_CHECK_INTERVAL_M = 0.05;
+// distance to destination before the path is considered to
+// have reached the destination
+static constexpr double DEST_CHECK_EPSILON_M = 1e-3;
+
+using namespace Test;
+
+using PathPlannerConstructor = std::function<std::unique_ptr<PathPlanner>()>;
+
+struct PlannerTestCase
+{
+    std::string name = "Unnamed test case";
+    Point start, dest;
+    Rectangle navigable_area;
+    std::vector<Obstacle> obstacles;
+    bool should_return_path;
+};
+
+std::vector<PlannerTestCase>
+    test_cases =
+        {{.name               = "Empty field straight line",
+          .start              = Point(0, 0),
+          .dest               = Point(1, 0),
+          .navigable_area     = Rectangle({-1, -1}, {1, 1}),
+          .obstacles          = {},
+          .should_return_path = true},
+
+         {.name           = "Single stationary robot in path",
+          .start          = Point(0, 0),
+          .dest           = Point(2, 0),
+          .navigable_area = Rectangle({-1, -1}, {1, 1}),
```

Comment:
navigable area doesn't contain start and dest

Reply:
fixed